phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexRebuildRegionScanner.java

```diff
@@ -298,69 +655,71 @@ private boolean checkIndexRow(final byte[] indexRowKey, final Put put) throws IO
     }
 
     private void logToIndexToolOutputTable(byte[] dataRowKey, byte[] indexRowKey, long dataRowTs, long indexRowTs,
-                                           String errorMsg) {
+                                           String errorMsg) throws IOException {
         logToIndexToolOutputTable(dataRowKey, indexRowKey, dataRowTs, indexRowTs,
                 errorMsg, null, null);
 
     }
 
     private void logToIndexToolOutputTable(byte[] dataRowKey, byte[] indexRowKey, long dataRowTs, long indexRowTs,
-                                           String errorMsg, byte[] expectedValue,  byte[] actualValue) {
+                                           String errorMsg, byte[] expectedValue,  byte[] actualValue) throws IOException {
         final byte[] E_VALUE_PREFIX_BYTES = Bytes.toBytes(" E:");
         final byte[] A_VALUE_PREFIX_BYTES = Bytes.toBytes(" A:");
         final int PREFIX_LENGTH = 3;
         final int TOTAL_PREFIX_LENGTH = 6;
+        final byte[] PHASE_BEFORE_VALUE = Bytes.toBytes("BEFORE");
+        final byte[] PHASE_AFTER_VALUE = Bytes.toBytes("AFTER");
+        long scanMaxTs = scan.getTimeRange().getMax();
+        byte[] keyPrefix = Bytes.toBytes(Long.toString(scanMaxTs));
+        byte[] rowKey;
+        if (dataRowKey != null) {
```

Comment:
Again, good to document what the PK schema is.

Reply:
I will do that